library/Vanilla/Web/ContentSecurityPolicy/DefaultContentSecurityPolicyProvider.php

```diff
@@ -52,10 +52,15 @@ private function getScriptSources(): array {
     */
     private function getFrameAncestors(): array {
         $scriptSrcPolicies[] = new Policy(Policy::FRAME_ANCESTORS, '\'self\'');
-        if ($whitelist = $this->config->get('Garden.TrustedDomains', false)) {
-            $trusteddDomains = is_string($whitelist) ? explode("\n", $whitelist) : [];
-            if (count($trusteddDomains) > 0) {
-                $scriptSrcPolicies[] = new Policy(Policy::FRAME_ANCESTORS, implode(' ', $trusteddDomains));
+        $whitelist = $this->config->get('Garden.TrustedDomains', false);
+        $trusteddDomains = is_string($whitelist) ? array_filter(explode("\n", $whitelist)) : [];
+        if (count($trusteddDomains) > 0) {
+            $scriptSrcPolicies[] = new Policy(Policy::FRAME_ANCESTORS, implode(' ', $trusteddDomains));
+        } else {
+            $remoteUrl = $this->config->get("Garden.Embed.RemoteUrl", false);
+            $remoteDomain = is_string($remoteUrl) ? parse_url($remoteUrl, PHP_URL_HOST) : false;
+            if (is_string($remoteDomain)) {
+                $scriptSrcPolicies[] = new Policy(Policy::FRAME_ANCESTORS, $remoteDomain);
```

Comment:
I think if got your idea correctly, we need to implement both `RemoteUrl` *and* `TrustedDomains` one after another but not `if ... else` which will be *either ... or*.
I can easily imagine situation when site configuration has both elements `RemoteUrl` and `TrustedDomains`.
Or do I miss something?